scripts/timelapse-colorizer-data/convert_emt_nuclear_data.py

```diff
@@ -0,0 +1,273 @@
+        # all rows in this group, since they are all on the same frame).
+        row = frame.iloc[0]
+        frame_number = row[TIMES_COLUMN]
+        # Flatten the z-stack to a 2D image.
+        zstackpath = row[SEGMENTED_IMAGE_COLUMN]
+        zstackpath = zstackpath.strip('"')
+        zstack = AICSImage(zstackpath).get_image_data("ZYX", S=0, T=0, C=0)
+        # Do a min projection instead of a max projection to prioritize objects which have lower IDs (which for this dataset,
+        # indicates lower z-indices). This is due to the nature of the data, where lower cell nuclei have greater confidence,
+        # and should be visualized when overlapping instead of higher nuclei.
+        # Do a min operation but ignore zero values. Without this, doing `min(0, id)` will always return 0 which results
+        # in black images. We use `np.ma.masked_equal` to mask out 0 values and have them be ignored,
+        # then replace masked values with 0 again (`filled(0)`) to get our final projected image.
+        masked = np.ma.masked_equal(zstack, 0, copy=False)
+        seg2d = masked.min(axis=0).filled(0)
+
+        # Scale the image and format as integers.
+        seg2d = scale_image(seg2d, scale)
+        seg2d = seg2d.astype(np.uint32)
+
+        # Remap the frame image so the IDs are unique across the whole dataset.
+        seg_remapped, lut = remap_segmented_image(
+            seg2d,
+            frame,
+            OBJECT_ID_COLUMN,
+        )
+
+        writer.write_image_and_bounds_data(
+            seg_remapped, grouped_frames, frame_number, lut
+        )
+
+        time_elapsed = time.time() - start_time
+        logging.info(
+            "Frame {} finished in {:5.2f} seconds.".format(
+                int(frame_number), time_elapsed
+            )
+        )
+
+
+def make_features(
+    dataset: pd.DataFrame,
+    features: List[str],
+    writer: ColorizerDatasetWriter,
+):
+    """
+    Generate the outlier, track, time, centroid, and feature data files.
+    """
+    # Collect array data from the dataframe for writing.
+
+    # For now in this dataset there are no outliers. Just generate a list of falses.
+    outliers = np.array([False for i in range(len(dataset.index))])
+    times = dataset[TIMES_COLUMN].to_numpy()
+    centroids_x = dataset[CENTROIDS_X_COLUMN].to_numpy()
+    centroids_y = dataset[CENTROIDS_Y_COLUMN].to_numpy()
+
+    # Generate the track data. This will be a very simple numpy table, where tracks[i] = i.
+    shape = dataset.shape
+    tracks = np.array([*range(shape[0])])
+
+    feature_data = []
+    for i in range(len(features)):
+        f = dataset[features[i]].to_numpy()
+        feature_data.append(f)
+
+    writer.write_feature_data(
+        feature_data,
+        tracks,
+        times,
+        centroids_x,
+        centroids_y,
+        outliers,
+    )
+
+
+def make_dataset(
+    data: pd.DataFrame,
+    output_dir="./data/",
+    dataset="3500005820_3",
+    do_frames=True,
```

Comment:
might be a stupid question, but what does `do_frames` mean?

Reply:
Not stupid! It's a flag, basically whether the script should export frame image data or skip it. If you just need to update feature/manifest data then you can skip regenerating the images.